osbs/build/build_request.py

```diff
@@ -732,6 +737,11 @@ def render(self, validate=True):
             self.spec.validate()
         super(ProductionBuild, self).render()
 
+        repo_name = git_repo_humanish_part_from_uri(self.spec.git_uri.value)
+        # What about forked repos?
```

Comment:
Oh, I see.

We don't really want to store the whole URI in the label, as labels also have a length limit we might hit.

Instead, we could say that different buildconfigs can share the same labels but will be distinguished by their spec.source.git.uri attributes.

I don't think this is a use-case we'll actually need to implement right now, so I'm fine with erroring out if there is already a build config with the same labels.

Later on if we need to we could handle the case that the labels query returns several build configs that match, and we have to further match on the spec.source.git.uri attribute.

What do you think?